lib/index.js

```diff
@@ -123,13 +123,33 @@ function updateCommand(entry) {
 }
 
 function needsUpdate(before, after) {
-  if (before.isDirectory() && after.isDirectory()) {
-    return false;
-  }
+  var invalidate;
 
-  var invalidate = before.size !== after.size ||
+  if (before.isDirectory() && after.isDirectory()) {
+    invalidate = false;
+  } else {
+    invalidate = before.size !== after.size ||
          +before.mtime !== +after.mtime ||
          before.mode !== after.mode;
+  }
+
+  if (!invalidate) {
+    var beforeMeta = before.meta;
+    var afterMeta = after.meta;
+    var beforeMetaKeys = beforeMeta ? Object.keys(beforeMeta) : [];
```

Comment:
@stefanpenner key order doesn't need to be stable: we check length and then we pick one arbitrarily (`beforeMeta`) and use its order